python/lib/dependabot/python/language_version_manager.rb

```diff
@@ -67,12 +66,10 @@ def python_version_from_supported_versions
           find { |v| requirement.satisfied_by?(Python::Version.new(v)) }
         return version if version
 
-        # If not, and we're dealing with a simple version string
-        # and changing the patch version would fix things, we do that
-        # as the patch version is unlikely to affect resolution
+        # If not, and we're dealing with a simple version string and changing the patch version would fix things,
+        # we do that as the patch version is unlikely to affect resolution
         if requirement_string.start_with?(/\d/)
-          requirement =
-            Python::Requirement.new(requirement_string.gsub(/\.\d+$/, ".*"))
```

Comment:
Oh interesting, this isn't actually truncating to `major.minor`, it's actually setting a wildcard/range:
```ruby
(ruby) z = Python::Version.new(requirement_string).truncate_to_major_minor
Gem::Version.new("3.6")
(ruby) zz = Python::Requirement.new(requirement_string.gsub(/\.\d+$/, ".*"))
Gem::Requirement.new(["~> 3.6.0.a"])
(ruby) requirement_string.gsub(/\.\d+$/, ".*")
"3.6.*"
```

Reply:
Oh silly me, `Python::Requirement` != `Python::Version` 🤦 

There goes two hours of my life. No wonder the truncate didn't work here.